src/provider/treeView.ts

```diff
@@ -119,7 +134,30 @@ export class MaestroWorkBenchTreeViewProvider implements vscode.TreeDataProvider
             }
         });
 
-        return filteredItems;
+        return finalItems;
+    }
+
+    private async hasMatchingFiles(folderPath: string): Promise<boolean> {
+        const entries = await fs.promises.readdir(folderPath, { withFileTypes: true });
```

Comment:
Given that we already have the filteredItems, do we need to go back to the file system?

Reply:
Updated in cc08df125e33343672f670fd118b679a0298749f